### ISSUE TITLE
Laravel: Classes without constructor sometimes return a null-constructor using Reflection

### ISSUE THREAD

**Issue by calvinbaart** [peachpiecompiler/peachpie#403]
Hi,

Laravel uses the Reflection API to provide Dependency Injection capabilities to classes. This happens through this code:
```php
        $reflector = new ReflectionClass($concrete);

        // If the type is not instantiable, the developer is attempting to resolve
        // an abstract type such as an Interface or Abstract Class and there is
        // no binding registered for the abstractions so we need to bail out.
        if (! $reflector->isInstantiable()) {
            return $this->notInstantiable($concrete);
        }

        $this->buildStack[] = $concrete;

        $constructor = $reflector->getConstructor();

        // If there are no constructors, that means there are no dependencies then
        // we can just resolve the instances of the objects right away, without
        // resolving any other types or dependencies out of these containers.
        if (is_null($constructor)) {
            array_pop($this->buildStack);

            return new $concrete;
        }

        $dependencies = $constructor->getParameters();
```

It seems that in some cases (can't reproduce it in a minimal example) the getConstructor call returns a constructor instance for classes without a user-defined constructor. This creates a null-pointer exception within getParameters. I'll try to create a sample repository this evening

**Comment by calvinbaart** [comment]
Alright, gotten up to making a basic laravel setup to demonstrate this issue: https://github.com/calvinbaart/laravel-peachpie-sample

The crash happens as soon as you try to load a page. It happens during the initialization phase in `website/vendor/calvinbaart/laravel-peachpie/src/Illuminate/Container/Container.php` around line 805.

I'll try to see if I can setup a development version of PeachPie so I can debug this issue.

**Comment by bfistein** [comment]
This is really awesome @calvinbaart! Thanks for all your well structured issues, exciting to see how close you are with such a massive project.

**Comment by calvinbaart** [comment]
Thanks, I'm just excited to get Laravel running in PeachPie :)

For some reason the debug builds I'm trying to make wont work but I figured out what's going wrong. The original code has this if statement:
```php
if (is_null($constructor)) {
```
Even though the debugger in PHP says that $constructor is null the is_null call never returns true. changing this to
```php
if ($constructor === null) {
```
also doesn't work.

After I changed it to:
```php
if (!$reflector->hasMethod("__construct")) {
```
it started to work properly.

It seems that the null-checks against ReflectionMethod fails for some reason even though the variable is NULL.
